### ISSUE TITLE
Alignment and Snap Problem in RulerPicker Component

### ISSUE THREAD

**Issue by abhay-rana** [rnheroes/react-native-ruler-picker#3]
I'm facing an alignment issue and snap problem with the RulerPicker component in the react-native-ruler-picker library. The teeth of the ruler picker are not aligned properly, and the snapping behavior is not accurate.

**Steps to reproduce**

1:) Include the RulerPicker component in your React Native project.

2:) <RulerPicker
  min={100}
  max={5000}
  step={100}
  fractionDigits={0}
  unit="ML"
  indicatorHeight={52}
  textAlign="bottom"
  gapBetweenSteps={5}
  indicatorColor="#D1FF70"
  shortStepColor="#D1FF70"
  longStepColor="#D1FF70"
  valueTextStyle={/* your value text style */}
  unitTextStyle={/* your unit text style */}
/>

3:) Observe the misalignment of the teeth and the snapping behavior.

Expected Behavior:

The teeth of the ruler picker should align perfectly with the bottom teeth.
When scrolling, the snapping behavior should accurately snap to the teeth positions.

![WhatsApp Image](https://github.com/rnheroes/react-native-ruler-picker/assets/41327466/5d6e884f-8464-4420-a6f7-98e22ba86025)

as you can seen in the image the teeth is not aligned vertically

React Native version: 0.72
Operating System: Windows 11
Device: Android 13

**Comment by hazimal** [comment]
same issue in my project

**Comment by enestatli** [comment]
@abhay-rana @hazimal only happening on Android?

**Comment by hazimal** [comment]
Tested only with android

**Comment by moulie415** [comment]
@Skipperlla @enestatli Is this being addressed? I can confirm I am experiencing this issue on android as well, iOS is fine

**Comment by ermapsh** [comment]
Yes I'm also facing the same issue, @rnheroes ![Screenshot_2024-07-04-12-00-52-306_com docexa goroga](https://github.com/rnheroes/react-native-ruler-picker/assets/72149385/02c15a62-9bd8-4a63-b44f-8d582bd07205)

**Comment by ermapsh** [comment]
> same issue in my project

how do you solve this?

**Comment by haerim95** [comment]
I'm having the same problem.  Did you come up with a solution?
Only on Android, it doesn't align and doesn't even scroll.

**Comment by haerim95** [comment]
> @abhay-rana @hazimal only happening on Android?

I'm having the same problem. Did you come up with a solution?
Only on Android, it doesn't align and doesn't even scroll.

**Comment by tungmin97** [comment]
This is an issue with how FlashList calculate layout on Android. On a similar list in our project, I have to use getLayoutManager to get the correct pixel distance for that specific index.

**Comment by haerim95** [comment]
> This is an issue with how FlashList calculate layout on Android. On a similar list in our project, I have to use getLayoutManager to get the correct pixel distance for that specific index.

This library currently uses AnimatedFlashList, not FlashList. I don't think AnimatedFlashList has a getLayoutManager option.

**Comment by tungmin97** [comment]
> > This is an issue with how FlashList calculate layout on Android. On a similar list in our project, I have to use getLayoutManager to get the correct pixel distance for that specific index.
> 
> This library currently uses AnimatedFlashList, not FlashList. I don't think AnimatedFlashList has a getLayoutManager option.

Ofc AnimatedFlashList doesn't have this prop,it comes from recyclerlistview, the inner list rendering engine. Anyway, here is the magic code that fixed my issue on Android, it does look very weird though.



    listRef.current?.scrollToOffset({
        offset: listRef?.current?.state?.layoutProvider
            ?.getLayoutManager()
            ?.getOffsetForIndex(index)?.x ?? index * ITEM_WIDTH,
        animated: true,
    });

**Comment by Bipinstacksmith** [comment]
I'm getting when I'm creating the releaseAPK after the install of apk , the slider is not working but it was working in when i'm running the project. How to solve this?

**Comment by carlo-rsd** [comment]
Did anyone get this to work on Android?
